src/guiapi/include/WindowItemFormatableLabel.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file WindowMenuSensor.hpp
+ * @author Vitezslav Machacek
+ * @brief Derived from WI_LABEL/IWindowMenuItem
+ * @date 2021-6-9
+ */
+
+#pragma once
+#include "WindowMenuLabel.hpp"
+#include <functional>
+
+/// This WI has lambda function for formatting the extension
+/// \tparam ValueType must have not equals and assigment operators and copy constructor
+template <class ValueType>
+class WI_FORMATABLE_LABEL_t : public AddSuper<WI_LABEL_t> {
+private:
+    static constexpr font_t *&InfoFont = GuiDefaults::FontMenuSpecial;
+    static constexpr uint16_t icon_width = 16;
+
+    const std::function<void(char *)> printAs;
+
+protected:
+    ValueType value;
+    ValueType oldVal;
+    constexpr static const char *NA = N_("N/A");
+    constexpr static const char *NI = N_("Not initialized");
+
+    void printExtension(Rect16 extension_rect, color_t color_text, color_t color_back, ropfn raster_op) const override {
```

Comment:
add non-template parent containing this method (and everything else not depended on ValueType)

Reply:
👍